validation/superchain-version_test.go

```diff
@@ -72,54 +70,48 @@ func getContractVersionsFromChain(list AddressList, client *ethclient.Client) (C
 	for _, contractAddress := range contractsToCheck {
 		a, err := list.AddressFor(contractAddress)
 		if err != nil {
-			return ContractVersions{}, err
+			return L1ContractBytecodeHashes{}, err
 		}
 		wg.Add(1)
-		go getVersionAsync(a, results, contractAddress, wg)
+		go getBytecodeHashAsync(a, results, contractAddress, wg)
 	}
 
 	wg.Wait()
 
-	// use reflection to convert results mapping into a ContractVersions object
+	// use reflection to convert results mapping into a L1ContractBytecodeHashes object
 	// without resorting to boilerplate code.
-	cv := ContractVersions{}
+	cbh := L1ContractBytecodeHashes{}
 	results.Range(func(k, v any) bool {
-		s := reflect.ValueOf(cv)
+		s := reflect.ValueOf(cbh)
 		for i := 0; i < s.NumField(); i++ {
 			// The keys of the results mapping come from the AddressList type,
 			// which includes both proxied and unproxied contracts.
-			// The cv object (of type ContractVersions), on the other hand,
-			// only lists implementation contract versions. The next line accounts for
-			// this: we may get the version directly from the implemntation, or via a Proxy,
+			// The cbh object (of type L1ContractBytecodeHashes), on the other hand,
+			// only lists implementation contract bytecode hashes. The next line accounts for
+			// this: we may get the bytecode hash directly from the implementation, or via a Proxy,
 			// but we store it against the implementation name in either case.
 			if s.Type().Field(i).Name == k || s.Type().Field(i).Name+"Proxy" == k {
-				reflect.ValueOf(&cv).Elem().Field(i).SetString(v.(string))
+				reflect.ValueOf(&cbh).Elem().Field(i).SetString(v.(string))
 			}
 		}
 		return true
 	})
 
-	return cv, nil
+	return cbh, nil
 }
 
-// getVersion will get the version of a contract at a given address, if it exposes a version() method.
-func getVersion(ctx context.Context, addr common.Address, client *ethclient.Client) (string, error) {
-	isemver, err := bindings.NewISemver(addr, client)
-	if err != nil {
-		return "", fmt.Errorf("%s: %w", addr, err)
-	}
-	version, err := Retry(isemver.Version)(&bind.CallOpts{
-		Context: ctx,
-	})
+// getBytecodeHash will get the hash of the bytecode of a contract at a given address.
+func getBytecodeHash(ctx context.Context, addr common.Address, client *ethclient.Client) (string, error) {
+	code, err := client.CodeAt(ctx, addr, nil)
```

Comment:
We need to change this so that _if_ the contract is proxied, we are getting the code of the _implementation_ contract, not the proxy.

Reply:
You can get the address of the proxy admin like this: 
```go
	Addresses[chain.ChainID].ProxyAdmin 
```